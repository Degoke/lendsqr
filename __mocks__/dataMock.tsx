import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
