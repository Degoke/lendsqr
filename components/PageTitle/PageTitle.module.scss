@use '../../styles/variables' as *;

.wrapper {
  h1 {
    font-weight: 500;
    font-size: 24px;
    color: $text-secondary;
  }

  &__button {
    display: flex;
    align-items: center;
    gap: 1rem;
  }
}
