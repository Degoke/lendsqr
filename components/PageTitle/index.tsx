import Image from 'next/image'
import Link from 'next/link'
import BackArrow from '../../public/images/back_arrow.svg'
import { urls } from '../../utils/constants/urls'
import styles from './PageTitle.module.scss'

type Props = {
  title: string
  buttonText?: string
}

export default function PageTitle({ title, buttonText }: Props) {
  return (
    <div className={styles.wrapper}>
      {buttonText && (
        <Link href={urls.USERS} className={styles.wrapper__button}>
          <Image src={BackArrow} alt={buttonText} />
          <span>{buttonText}</span>
        </Link>
      )}
      <h1>{title}</h1>
    </div>
  )
}
