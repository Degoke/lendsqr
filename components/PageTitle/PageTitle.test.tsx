import { render, screen } from '@testing-library/react'
import PageTitle from './index'
import { mockCard, mockImage } from '../../__mocks__/dataMock'

describe('PageTitle tests', () => {
  it('renders PageTitle component unchanged', () => {
    const { container } = render(<PageTitle title={'Hello'} />)
    expect(container).toMatchSnapshot()
  })
})
