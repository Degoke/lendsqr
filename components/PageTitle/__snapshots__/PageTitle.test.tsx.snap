// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle tests renders PageTitle component unchanged 1`] = `
<div>
  <div
    class="wrapper"
  >
    <h1>
      Hello
    </h1>
  </div>
</div>
`;
